src/server/shell/decoration/basic_manager.cpp

```diff
@@ -85,3 +139,30 @@ void msd::BasicManager::undecorate_all()
     // Destroy the decorations outside the lock
     to_destroy.clear();
 }
+
+void msd::BasicManager::advise_output_update(mg::DisplayConfiguration const& config)
+{
+    // Use the maximum scale to ensure sharp-looking decorations on all outputs
+    auto max_output_scale{0.0f};
+    config.for_each_output(
+        [&](mg::DisplayConfigurationOutput const& output)
+        {
+            if (!output.used || !output.connected) return;
+            if (!output.valid() || (output.current_mode_index >= output.modes.size())) return;
+
+            max_output_scale = std::max(max_output_scale, output.scale);
+        });
+
+    if (max_output_scale == scale) return;
+
+    scale = max_output_scale;
+
+    std::lock_guard lock{mutex};
+    for (auto& it : decorations)
+    {
+        if (it.second)
```

Comment:
I don't think this condition can fail: an empty decoration should not be possible